src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -95,10 +101,11 @@ public abstract class SqlMetadataProvider<ConnectionT, DataAdapterT, CommandT> :
                 }
             }
 
-            ConnectionString = runtimeConfig.DataSource.ConnectionString;
+            ConnectionString = runtimeConfig.GetDataSourceFromDataSourceName(dataSourceName).ConnectionString;
             EntitiesDataSet = new();
-            SqlQueryBuilder = queryBuilder;
-            QueryExecutor = queryExecutor;
+            EngineFactory = engineFactory;
```

Comment:
The properties `SqlQueryBuilder`, `QueryExecutor` can be instantiated inside the respective MetadataProvider classes for Pg/My/Ms sql.

Also a nit (If we want to use `EngineFactory` to instantiate them still) : Should we rename the class to something like `AbstractEngineFactory`? Going by definition, this follows abstract factory pattern.